packages/core/tsconfig.json

```diff
@@ -20,5 +9,25 @@
     {
       "path": "./tsconfig.spec.json"
     }
-  ]
+  ],
+  "compilerOptions": {
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "jsx": "preserve",
+    "allowJs": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "types": [
+      "node",
+      "jest"
+    ],
+    "noEmit": true,
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "incremental": true,
+    "noImplicitOverride": true,
+    "noPropertyAccessFromIndexSignature": true,
+  }
```

Comment:
Not had a lot of time to study these different options, but we might want to consider what the implications are.
- Maybe some of these are defaults, but others may conflict with our standards such as `allowJs`.
- Previously we had `"jsx": "react-jsx"`, now it's `"jsx": "preserve"` - not sure what that means for us